internal/servers/worker/status.go

```diff
@@ -115,6 +159,33 @@ func (w *Worker) sendWorkerStatus(cancelCtx context.Context) {
 	})
 	if err != nil {
 		w.logger.Error("error making status request to controller", "error", err)
+		// Check for last successful status. Ignore nil last status, this probably
+		// means that we've never connected to a controller, and as such probably
+		// don't have any sessions to worry about anyway.
+		//
+		// If a length of time has passed since we've been able to communicate, we
+		// want to start terminating all sessions as a "break glass" kind of
+		// scenario, as there will be no way we can really tell if these
+		// connections should continue to exist.
+
+		lastStatus := w.LastStatusSuccess()
```

Comment:
I think it's important to capture something more for when lastStatus is nil. It seems _unlikely_ that we'd never have sent status but have connections, but I think what we could do is on worker startup time capture the current time. If lastStatus is nil, use that instead as the comparison for the grace period.

Reply:
Fixed!